### ISSUE TITLE
"Use an interface instead of a type literal" warning even though type cannot be replaced by interface

### ISSUE THREAD

**Issue by letorbi** [palantir/tslint#2662]
### Bug report

- __TSLint version__:  4.5.1 (tslint-loader: 3.4.3, tslint-react: 2.5.0)
- __TypeScript version__: 2.2.1

**Code**

*test-inline.ts*
```ts
interface IKeyValue {
  [key: string]: string;
}

interface IParent {
  foo?: IKeyValue;
  goo?: string;
}

// works

export interface ITestA extends IParent {
  foo?: {
    bar: string,
  };
}

// warning

type TFoo = {
  bar: string,
};

export interface ITestB extends IParent {
  foo?: TFoo;
}

// error
/*
interface IFoo {
    bar: string;
};

interface ITestC extends IParent {
  foo: IFoo;
}
*/

export class TestInline {
  public testA: ITestA;
  public testB: ITestB;
  // public testC: ITestC;

  public foo() {
    return 'foo';
  }
}
```

*anywhere else*
```ts
import {TestInline} from 'test-inline';

const ti = new TestInline();
```

**Config**

```
{
  "extends": ["tslint:latest", "tslint-react"],
  "rules": {
    // not in tslint:latest yet
    "no-unused-variable": [true, "react"],
    // as we are all used to prefer single quotes
    "quotemark": [ true, "single", "jsx-single", "avoid-escape" ],

    // also allow leading underscore
    "variable-name": {
        "options": [
            "ban-keywords",
            "check-format",
            "allow-leading-underscore",
            "allow-pascal-case"
        ]
    }
  }
}
```

**Expected behavior:**

If the warning *"Use an interface instead of a type literal"* is thrown, I should be able to replace the type literal by an interface without causing an error. If `type` and `interface` are not equivalent such a warning should not be thrown.

**Actual behavior:**

When I'm replacing the type literal at *interface.ts* line 19 with an interface definition (see commented out code), the compiler throws an error: *[33:11] Interface 'ITestC' incorrectly extends interface `IParent` [...] Index signature is missing in type 'IFoo'*.

**Comment by ajafff** [comment]
The interface and the type literal should be handled the same by the typescript compiler. I would consider this a bug in typescript, not in tslint. There's already an open issue: https://github.com/Microsoft/TypeScript/issues/15300

**Comment by letorbi** [comment]
Yeah you're right. In the end it should be fixed in TypeScript...
